packages/blocks/src/_common/components/block-caption.ts

```diff
@@ -138,17 +138,26 @@ export class BlockCaptionEditor<
     ></textarea>`;
   }
 
-  @property({ attribute: false })
-  accessor block!: BlockComponent<Model> & { isInSurface?: boolean };
-
   @state()
   accessor caption: string | null | undefined = undefined;
 
   @state()
   accessor display = false;
 
+  @property({ attribute: false })
+  accessor doc!: Doc;
+
+  @property({ attribute: false })
+  accessor host!: EditorHost;
+
   @query('.block-caption-editor')
   accessor input!: HTMLInputElement;
+
+  @property({ attribute: true })
+  accessor mode!: 'edgeless' | 'page';
+
+  @property({ attribute: false })
+  accessor model!: Model;
```

Comment:
We also have the `model` context here.
https://github.com/toeverything/blocksuite/blob/master/packages/framework/block-std/src/view/element/block-component.ts#L332